docs/data/data-grid/charts-integration/GridChartsIntegrationCustomization.tsx

```diff
@@ -0,0 +1,273 @@
+      id: `${i}-${j}`,
+      timestamp: new Date(downloads.timestamps[i]),
+      version: versions[j],
+      downloads: downloads.versionDownloads[versions[j]][i],
+    });
+  }
+}
+
+const hideColorsControl = (sections: GridChartsConfigurationSection[]) =>
+  sections.map((section) => ({
+    ...section,
+    controls: {
+      ...section.controls,
+      colors: {
+        ...section.controls.colors,
+        isHidden: () => true,
+      },
+    },
+  }));
+
+const customConfiguration = {
+  bar: {
+    ...configurationOptions.bar,
+    customization: hideColorsControl(configurationOptions.bar.customization),
+    maxCategories: 1,
+  },
+  column: {
+    ...configurationOptions.column,
+    customization: hideColorsControl(configurationOptions.column.customization),
+    maxCategories: 1,
+  },
+  line: {
+    ...configurationOptions.line,
+    customization: hideColorsControl(configurationOptions.line.customization),
+    maxCategories: 1,
+  },
+  area: {
+    ...configurationOptions.area,
+    customization: hideColorsControl(configurationOptions.area.customization),
+    maxCategories: 1,
+  },
+};
+
+const gridPivotModel: GridPivotModel = {
+  rows: [{ field: 'timestamp' }],
+  columns: [{ field: 'majorVersion', sort: 'asc' }],
+  values: [{ field: 'downloads', aggFunc: 'sum' }],
+};
+
+const getPivotDerivedColumns: DataGridPremiumProps['getPivotDerivedColumns'] = (
+  column,
+) => {
+  if (column.field === 'version') {
+    return [
+      {
+        field: 'majorVersion',
+        headerName: `Major version`,
+        type: 'number',
+        valueGetter: (_, row) => Number(row.version.split('.')[0]),
+        valueFormatter: (value: string) => `v${value}`,
+      },
+    ];
+  }
+  return undefined;
+};
+
+const initialState = {
+  sidebar: {
+    open: true,
+    value: GridSidebarValue.Charts,
+  },
+  pivoting: {
+    enabled: true,
+    model: gridPivotModel,
+  },
+  chartsIntegration: {
+    charts: {
+      main: {
+        categories: ['timestamp'],
+        series: [],
+        chartType: 'line',
+        configuration: {
+          showMark: false,
+          height: 400,
+        },
+      },
+    },
+  },
+};
+
+const getColumnName: GridChartsPanelProps['getColumnName'] = (field) => {
+  if (field === 'downloads') {
+    return 'Downloads';
+  }
+  if (!field.endsWith('downloads')) {
+    return undefined;
+  }
+  return `v${field[0]}`;
+};
+
+const dateFormatter = (value: string | Date) =>
+  new Date(value).toLocaleDateString('en-US', {
+    month: '2-digit',
+    year: '2-digit',
+  });
+const downloadsFormatter = (value: number) =>
+  value === 0 ? '0' : `${Math.round(value / 1000)}k`;
+
+const onRender = (
+  type: string,
+  props: Record<string, any>,
+  Component: React.ComponentType<any>,
+) => {
+  let adjustedProps = props;
+
+  if (type === 'line' || type === 'area') {
+    adjustedProps = {
+      ...adjustedProps,
+      grid: {
+        vertical: true,
+        horizontal: true,
+      },
```

Comment:
I'm not sure I follow the need for the `onRender` prop. I was trying to understand why changing the grid has no effect on the line chart while it does on the bar chart for this example 😅

Maybe we can release it like this and see how users react? Though I would suspect having a "default values" would be better, at least for this use-case.

Reply:
Nvm there is a default state, I've moved it there 🤣